src/main/java/uk/gov/pay/ledger/transactionsummary/dao/TransactionSummaryDao.java

```diff
@@ -0,0 +1,77 @@
+package uk.gov.pay.ledger.transactionsummary.dao;
+
+import com.google.inject.Inject;
+import org.jdbi.v3.core.Jdbi;
+import uk.gov.pay.ledger.transaction.state.TransactionState;
+
+import java.time.ZonedDateTime;
+
+public class TransactionSummaryDao {
+
+    private static final String UPSERT_STRING = "INSERT INTO transaction_summary AS ts(gateway_account_id, type, " +
+            " transaction_date, state, live, moto, total_amount_in_pence, no_of_transactions, total_fee_in_pence)" +
+            " VALUES(:gatewayAccountId, :type, :transactionDate, :state, :live, :moto, :amountInPence, 1,:feeInPence)" +
+            " ON CONFLICT ON CONSTRAINT transaction_summmary_unique_key " +
+            " DO UPDATE SET no_of_transactions = ts.no_of_transactions + 1, " +
+            " total_amount_in_pence = :amountInPence + ts.total_amount_in_pence," +
+            " total_fee_in_pence = :feeInPence + ts.total_fee_in_pence" +
+            " WHERE ts.gateway_account_id = :gatewayAccountId " +
+            " AND ts.transaction_date = :transactionDate " +
+            " AND ts.state = :state " +
+            " AND ts.type = :type" +
+            " AND ts.live = :live" +
+            " AND ts.moto = :moto";
+
+    private static final String UPDATE_SUMMARY = "UPDATE transaction_summary ts " +
```

Comment:
Very minor but would it make sense to change the name of this constant to something like DEDUCT_FOR_ROW or something that's a little more descriptive? Super minor though so up to you.

Reply:
Good spot! I will update in next PR